src/ds/ds6/hw_monitor_extended_buffers.cpp

```diff
@@ -2,31 +2,22 @@
 // Copyright(c) 2023 Intel Corporation. All Rights Reserved.
 
 #include "hw_monitor_extended_buffers.h"
-
 #include "ds/ds6/ds6-private.h"
 
+
 namespace librealsense
 {
     int hw_monitor_extended_buffers::get_msg_length(command cmd) const
     {
         int msg_length = HW_MONITOR_BUFFER_SIZE;
         if (cmd.cmd == ds::fw_cmd::GET_HKR_CONFIG_TABLE || cmd.cmd == ds::fw_cmd::SET_HKR_CONFIG_TABLE)
         {
-            switch (cmd.param2)
-            {
-            case static_cast<int>(ds::ds6_calibration_table_id::coefficients_table_id):
-            {
-                msg_length = sizeof(ds::ds6_coefficients_table);
-                break;
-            }
-            case static_cast<int>(ds::ds6_calibration_table_id::rgb_calibration_id):
-            {
-                msg_length = sizeof(ds::ds6_rgb_calibration_table);
-                break;
-            }
-            default:
-                throw std::runtime_error(rsutils::string::from() << "command GET_HKR_CONFIG_TABLE with wromg param2");
-            }
+            auto calib_table_id = static_cast<ds::ds6_calibration_table_id>(cmd.param2);
+            auto it = ds::ds6_calibration_tables_size.find(calib_table_id);
+            if (it == ds::ds6_calibration_tables_size.end())
+                throw std::runtime_error(rsutils::string::from() << " hwm command with wromg param2");
+
+            msg_length = ds::ds6_calibration_tables_size.at(calib_table_id);
```

Comment:
We already have it, no need to search again.
```
msg_length = *it;
```

Reply:
right